frame/support/src/storage/generator/map.rs

```diff
@@ -207,6 +207,43 @@ where
 			}
 		}
 	}
+
+	fn translate_next<O: Decode, F: FnMut(K, O) -> Option<V>>(
```

Comment:
if this get merged, should we rewrite `translate` to use this internally?

Reply:
Why not add a `limit` to `translate`?

Reply:
> if this get merged, should we rewrite translate to use this internally?

I don't see a reason why not.

> Why not add a `limit` to `translate`?

A limit alone is not enough because we need to tell the function the last key key we translated in order to make progress. We also only implement the specific case (`limit = 1`) because our migration steps should have a fixed benchmarkable weight. Hence we only do a single translation per step. That makes the individual migration code simpler while shifting the complexity to the migration infrastructure code which drives the migration to completion.